### DISCUSSION TITLE
Help regarding Inferencing from UNETR/BTCV model

### DISCUSSION THREAD

**Discussion by pranavPrakhar** [Project-MONAI/MONAI#7687] [Q&A] [answered]
Hi everyone , I have trained a UNETR model and now I want to get prediction on many nifti files, and want the segmentation result to be stored. Is there already a script for that, since I was not able to find one?
I also have one more query, I wanted to know the image name of the image on which the prediction is being done. In [research-contributions](https://github.com/Project-MONAI/research-contributions/tree/main)/[UNETR](https://github.com/Project-MONAI/research-contributions/tree/main/UNETR)/[BTCV](https://github.com/Project-MONAI/research-contributions/tree/main/UNETR/BTCV/)test.py line 99 (screenshot attached) apparently does the same thing, but it is giving error:
"KeyError: 'image_meta_dict' ", upon printing all the keys of 'batch", there is no key named 'image_meta_dict'. Any help on this is highly appreciated.
<img width="747" alt="Screenshot 2024-04-20 at 11 28 51 PM" src="https://github.com/Project-MONAI/MONAI/assets/145749656/14d765c3-a28d-4254-9011-4fe925bf5787">

**Comment by KumoLiu** [comment]
Hi @pranavPrakhar, you can refer to this tutorial: https://github.com/Project-MONAI/tutorials/blob/main/3d_segmentation/unetr_btcv_segmentation_3d.ipynb.

Additionally, you can (1) add `image_only=False` in `LoadImage` or (2) use `batch['image'].meta` to get the meta.
For more information, please refer to https://github.com/Project-MONAI/MONAI/wiki/MetaTensor-guide

Hope it helps, thanks.

**Reply by pranavPrakhar** [reply, marked as answer]
Thanks @KumoLiu
